### PR TITLE
[16.0][MIG] sale_report_delivered: Migration to version 16.0

### PR DESCRIPTION
**Pull request by Yadier-Tecnativa**
@Tecnativa TT42436

### CODE REVIEW

File: sale_report_delivered/tests/test_sale_report_delivered.py [hunk 1]
```diff
@@ -83,4 +89,4 @@ def test_sale_report_delivered_read_group(self):
             ],
             groupby=["order_id"],
         )
-        self.assertEqual(res[0]["margin_percent"], 100)
+        self.assertEqual(round(res[0]["margin_percent"], 2), 100.00)
```
Comment (chienandalu) [inline sale_report_delivered/tests/test_sale_report_delivered.py]: Better use `assertAlmostEqual` for rounding issues

File: sale_report_delivered/tests/test_sale_report_delivered.py [hunk 2]
```diff
@@ -70,7 +76,7 @@ def test_sale_report_delivered_misc(self):
         self.assertIn(self.product, items.mapped("product_id"))
         self.assertNotIn(self.service, items.mapped("product_id"))
 
-    @users("admin", "test_user-sale_report_delivered")
+    @users("test_user-sale_report_delivered")
```
Comment (chienandalu) [inline sale_report_delivered/tests/test_sale_report_delivered.py:96]: Why removing `admin` from the test?

File: sale_report_delivered/tests/test_sale_report_delivered.py (hunk 1 above)
Comment (Yadier-Tecnativa) [reply sale_report_delivered/tests/test_sale_report_delivered.py]: Changed

File: sale_report_delivered/tests/test_sale_report_delivered.py (hunk 2 above)
Comment (Yadier-Tecnativa) [reply sale_report_delivered/tests/test_sale_report_delivered.py:96]: changed for user admin only

**Comment by Yadier-Tecnativa** [comment]
@sergio-teruel

File: sale_report_delivered/tests/test_sale_report_delivered.py [hunk 3]
```diff
@@ -70,7 +75,7 @@ def test_sale_report_delivered_misc(self):
         self.assertIn(self.product, items.mapped("product_id"))
         self.assertNotIn(self.service, items.mapped("product_id"))
 
-    @users("admin", "test_user-sale_report_delivered")
+    @users("admin")
```
Comment (chienandalu) [inline sale_report_delivered/tests/test_sale_report_delivered.py]: If the decorator doesn't work properly on the test for sereral users you can run two separate tests at least

Comment (Yadier-Tecnativa) [reply sale_report_delivered/tests/test_sale_report_delivered.py]: Changed

File: sale_report_delivered/tests/test_sale_report_delivered.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                "margin_percent:sum",
+                "price_subtotal:sum",
+                "margin:sum",
+            ],
+            groupby=["order_id"],
+        )
+        self.assertAlmostEqual(res[0]["margin_percent"], 100.00)
+
+    @users("test_user-sale_report_delivered")
+    def test_sale_report_delivered_read_group(self):
```
Comment (chienandalu) [inline sale_report_delivered/tests/test_sale_report_delivered.py:97]: better wrap the content in a common method a trigger it from both tests

File: sale_report_delivered/tests/test_sale_report_delivered.py [hunk 5]
```diff
@@ -70,10 +75,9 @@ def test_sale_report_delivered_misc(self):
         self.assertIn(self.product, items.mapped("product_id"))
         self.assertNotIn(self.service, items.mapped("product_id"))
 
-    @users("admin", "test_user-sale_report_delivered")
-    def test_sale_report_delivered_read_group(self):
+    def sale_report_delivered_read_group(self):
```
Comment (chienandalu) [inline sale_report_delivered/tests/test_sale_report_delivered.py]:
```suggestion
    def _test_sale_report_delivered_read_group(self):

File: sale_report_delivered/tests/test_sale_report_delivered.py (hunk 4 above)
Comment (Yadier-Tecnativa) [reply sale_report_delivered/tests/test_sale_report_delivered.py:97]: Solved @chienandalu

File: sale_report_delivered/tests/test_sale_report_delivered.py [hunk 6]
```diff
@@ -83,4 +87,13 @@ def test_sale_report_delivered_read_group(self):
             ],
             groupby=["order_id"],
         )
-        self.assertEqual(res[0]["margin_percent"], 100)
```
Comment (chienandalu) [inline sale_report_delivered/tests/test_sale_report_delivered.py]: Just leave everythin as it was and just call it from the decorated tests

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: sale_report_delivered/tests/test_sale_report_delivered.py (hunk 6 above)
Comment (Yadier-Tecnativa) [reply sale_report_delivered/tests/test_sale_report_delivered.py]: Done

**Review by sergio-teruel — APPROVED** [review]
Functional and code review

**Review by chienandalu — APPROVED** [review]

**Pull request merged**